trieste/bayesian_optimizer.py

```diff
@@ -84,7 +86,57 @@ def model(self) -> TrainableProbabilisticModel:
         if len(self.models) == 1:
             return next(iter(self.models.values()))
         else:
-            raise ValueError(f"Expected a single dataset, found {len(self.datasets)}")
+            raise ValueError(f"Expected a single model, found {len(self.models)}")
+
+    def save(self, path: Path) -> FrozenRecord[StateType]:
+        """Save the record to disk."""
+        path.parent.mkdir(exist_ok=True, parents=True)
+        with open(path, "wb") as f:
+            dill.dump(self, f, dill.HIGHEST_PROTOCOL)
+        return FrozenRecord(path)
+
+
+@dataclass(frozen=True)
+class FrozenRecord(Generic[StateType]):
+    """
+    A Record container saved on disk.
+
+    Note that records are saved via pickling and are therefore neither portable nor secure.
+    Only open frozen records generated on the same system.
+    """
+
+    path: Path
+    """ The path to the pickled Record. """
+
+    def load(self) -> Record[StateType]:
+        """Load the record into memory."""
+        with open(self.path, "rb") as f:
+            return dill.load(f)
```

Comment:
Whats dill? is it the same as pickle?

Reply:
Yes, dill is an extension of pickle that supports more things out of the box (e.g. lambdas).